### PR TITLE
GG-18932 Fixed errors in demo mode without agent.

### PR DESCRIPTION
**Pull request by vsisko**

### CODE REVIEW

File: modules/web-console/frontend/app/components/page-queries/components/queries-notebook/controller.ts [hunk 1]
```diff
@@ -963,19 +963,24 @@ export class NotebookCtrl {
...
-                    this.clusterIsAvailable = (!!cluster && cluster.active === true) || agentMgr.isDemoMode();
+            );
+
+            const checkState$ = combineLatest(
+                cluster$,
+                agentMgr.currentCluster$.pipe(pluck('state'), distinctUntilChanged())
+            ).pipe(
+                tap(([cluster, state]) => {
+                    if (state === 'CONNECTED')
+                        this.clusterIsAvailable = (!!cluster && cluster.active === true) || agentMgr.isDemoMode();
```
Comment (Klaster1) [inline modules/web-console/frontend/app/components/page-queries/components/queries-notebook/controller.ts]: I can't find any uses of `clusterIsAvailable` 🤷 Maybe remove the `tap` completely?

File: modules/web-console/frontend/app/components/page-queries/components/queries-notebook/controller.ts [hunk 2]
```diff
@@ -963,19 +963,24 @@ export class NotebookCtrl {
...
-                switchMap((cluster) => {
-                    if (!cluster && !agentMgr.isDemoMode()) {
-                        return of(EMPTY).pipe(
-                            tap(() => {
-                                $scope.caches = [];
-                            })
-                        );
+            this.refresh$ = checkState$.pipe(
+                switchMap(([cluster, state]) => {
+                    if (state !== 'CONNECTED' || (!cluster && !agentMgr.isDemoMode())) {
```
Comment (Klaster1) [inline modules/web-console/frontend/app/components/page-queries/components/queries-notebook/controller.ts]: I'd move side-effects into tap before `switchMap`.

Comment (vsisko) [reply modules/web-console/frontend/app/components/page-queries/components/queries-notebook/controller.ts]: Side effect moved to tap function.

File: modules/web-console/frontend/app/components/page-queries/components/queries-notebook/controller.ts (hunk 1 above)
Comment (vsisko) [reply modules/web-console/frontend/app/components/page-queries/components/queries-notebook/controller.ts]: Removed unused variable with tap function.

**Comment by vsisko** [comment]
Implemted e2e test

**Pull request merged**